src/use_case/kill_villager/KillVillagerInteractor.java

```diff
@@ -1,2 +1,40 @@
-package use_case.kill_villager;public class KillVillagerInteractor {
-}
+package use_case.kill_villager;
+
+import entity.Game;
+import entity.PromptGenerator;
+
+public class KillVillagerInteractor implements KillVillagerInputBoundary{
+
+    final KillVillagerOutputBoundary killPresenter;
+    final PromptGenerator gpt;
+    // We need to have this as one of the attributes since the PromptGenerator used throughout the application must
+    // contain the same conversation history, and whenever we use the PromptGenerator, we are adding to the conversation
+    // history.
+    final Game game;
+    // Originally put it as a parameter for killVillager, but it seems to make more sense to put it here since the
+    // KillVillagerOutputBoundary should only be handling one Game entity at a time.
+    final String villager;
+    // Yeah it should be here because the input data should handle everything.
+
+    public KillVillagerInteractor(KillVillagerOutputBoundary killPresenter,
+                                  KillVillagerInputData killVillagerInputData) {
+        this.killPresenter = killPresenter;
+        this.gpt = killVillagerInputData.getGpt();
+        this.game = killVillagerInputData.getGame();
+        this.villager = killVillagerInputData.getVillager();
+    }
+
+    @Override
+    public void killVillager() {
+        if (game.getAliveVillagers().containsKey(villager)) {
+            game.killPlayer(villager);
+            // Here I am assuming that the game entity will properly update the aliveVillagers hashmap and properly
+            // update the deadVillagers hashmap
+            game.changeGameState();
```

Comment:
i'll adjust my use case to change state as well perhaps?

Reply:
I know we already talked about this but I'm pretty sure String villager shouldn't be an attribute. The input data is supposed to be passed into killVillager(InputData input data) <-- like that, and the way this is right now your input data class doesn't do anything. I don't think your output data does anything currently either.

Reply:
I didn't use an input data or output data because for my case, both would only be containing one string. I think Derek said it's fine to not implement the input/output data if we don't think it's necessary as long as it's justified.

Reply:
I can modify the code so it uses input/output data if you want